src/i_midimusic.c

```diff
@@ -1674,82 +1413,74 @@ static void I_WIN_SetMusicVolume(int volume)
     }
     last_volume = volume;
 
-    EnterCriticalSection(&CriticalSection);
-    volume_factor = sqrtf((float)volume / 15);
-    update_volume = (song.file != NULL);
-    LeaveCriticalSection(&CriticalSection);
+    SDL_LockMutex(music_lock);
+    volume_factor = sqrtf((float)volume / 15.0f);
+    if (song.file)
+    {
+        UpdateVolume();
+    }
```

Comment:
This was changed so it calls directly from the main thread instead of setting a state for later in the midi thread. Now something in i_sound.c or s_sound.c changes the volume when Woof launches (but before `ResetDevice()` is called) and is setting the volume to zero needlessly.

<details><summary>MIDI-OX output</summary>

Before:
```
 TIMESTAMP IN PORT STATUS DATA1 DATA2 CHAN NOTE EVENT               
 000016D4   3  --     F0  Buffer:     6 Bytes   System Exclusive      
 SYSX: F0 7E 7F 09 01 F7
[other reset messages]
```

After:
```
 TIMESTAMP IN PORT STATUS DATA1 DATA2 CHAN NOTE EVENT               
 00006D42   3  --     B0    07    00    1  ---  CC: Volume            
 00006D42   3  --     B1    07    00    2  ---  CC: Volume            
 00006D42   3  --     B2    07    00    3  ---  CC: Volume            
 00006D42   3  --     B3    07    00    4  ---  CC: Volume            
 00006D42   3  --     B4    07    00    5  ---  CC: Volume            
 00006D42   3  --     B5    07    00    6  ---  CC: Volume            
 00006D42   3  --     B6    07    00    7  ---  CC: Volume            
 00006D42   3  --     B7    07    00    8  ---  CC: Volume            
 00006D42   3  --     B8    07    00    9  ---  CC: Volume            
 00006D43   3  --     B9    07    00   10  ---  CC: Volume            
 00006D43   3  --     BA    07    00   11  ---  CC: Volume            
 00006D43   3  --     BB    07    00   12  ---  CC: Volume            
 00006D43   3  --     BC    07    00   13  ---  CC: Volume            
 00006D43   3  --     BD    07    00   14  ---  CC: Volume            
 00006D43   3  --     BE    07    00   15  ---  CC: Volume            
 00006D43   3  --     BF    07    00   16  ---  CC: Volume   
 00006D43   3  --     F0  Buffer:     6 Bytes   System Exclusive      
 SYSX: F0 7E 7F 09 01 F7
[other reset messages]
```

</details>

Reply:
Fixed in https://github.com/fabiangreffrath/woof/pull/1635/commits/5b182c593c8aa63069eeb64203a9536d08a9565a